Anki.Vector/Robot.cs

```diff
@@ -537,6 +537,7 @@ private async Task ConnectToChannel(Task connectionTask)
             catch (Exception ex)
             {
                 // If failed to open channel throw exception
+                Dispose();
```

Comment:
You can't dispose() here or you will break clients that hold a robot reference and use the connect and disconnect methods.  They'll have their live object disposed of when connect fails and that's not good.  It's not clear to me what you're trying to stop from happening here.

Reply:
I think you need to move this to the NewConnection() method that does the connection and dispose the robot instance created there on failure.

Reply:
But I'm curious what it means that the connection continues in the background?  It should just do each connection one at a time and it waits for completion.  On failure does something else weird happen?

I'm thinking maybe it would be better to parallelize the connections -- attempt to connect to all Vector's at the same time and just keep the one that finishes first.  Then you don't have to wait for the timeout to finish on each one before trying the next.  I don't have more than one Vector so I can't help too much with testing this.

Reply:
When I have NewConnection() called for the first robot in the configuration file, and it is off, moves onto the NewConnect() for the the next connection.  In the background the GRPC keeps trying to connect to the 1st robot.  (This could be useful in other contexts, but not here, since the object reference is gone).  This can be most easily observed by setting GRPC_VERBOSITY to "DEBUG" and it will spew a lot of log messages every few seconds.

It could be my background, but I find it better to get the serial case (of some task) working correctly first, then make the parallelized version... A parallel connect doesn't seem to be much value right now.  Serial connect isn't very slow (VS2019 start up costs are though)

Reply:
>  In the background the GRPC keeps trying to connect to the 1st robot. 

Weird!  That is non-intuitive behavior.  I think in addition to the change you made to NewConnection(), the actual Connect() method needs to do some cleanup on failure.  Basically calling `channel.ShutdownAsync()` if it didn't connect.  I will add that.